core/src/Kokkos_MathematicalSpecialFunctions.hpp

```diff
@@ -1268,6 +1269,304 @@ KOKKOS_INLINE_FUNCTION CmplxType cyl_bessel_h21(const CmplxType& z) {
+    const IntType& a, const Kokkos::complex<RealType>& z) {
+  return Impl::cyl_bessel_j<Kokkos::complex<RealType>, RealType, IntType>(a, z);
+}
+
+template <class RealType, class IntType>
+KOKKOS_INLINE_FUNCTION Kokkos::complex<RealType> cyl_bessel_y(
+    const IntType& a, const Kokkos::complex<RealType>& z) {
+  return Impl::cyl_bessel_y<Kokkos::complex<RealType>, RealType, IntType>(a, z);
+}
+
+template <class RealType, class IntType>
+KOKKOS_INLINE_FUNCTION Kokkos::complex<RealType> cyl_bessel_h1(
+    const IntType& a, const Kokkos::complex<RealType>& z) {
+  return Impl::cyl_bessel_h1<Kokkos::complex<RealType>, RealType, IntType>(a,
+                                                                           z);
+}
+
+template <class RealType, class IntType>
+KOKKOS_INLINE_FUNCTION Kokkos::complex<RealType> cyl_bessel_h2(
+    const IntType& a, const Kokkos::complex<RealType>& z) {
+  return Impl::cyl_bessel_h2<Kokkos::complex<RealType>, RealType, IntType>(a,
+                                                                           z);
+}
+
+template <class RealType, class IntType>
+KOKKOS_INLINE_FUNCTION Kokkos::complex<RealType> cyl_bessel_i(
+    IntType a, const Kokkos::complex<RealType>& z) {
+  return Impl::cyl_bessel_i<Kokkos::complex<RealType>, RealType, IntType>(a, z);
+}
+
+template <class RealType, class IntType>
+KOKKOS_INLINE_FUNCTION Kokkos::complex<RealType> cyl_bessel_k(
+    IntType a, const Kokkos::complex<RealType>& z) {
+  return Impl::cyl_bessel_k<Kokkos::complex<RealType>, RealType, IntType>(a, z);
+}
+
+template <typename RealType, typename IntType>
+KOKKOS_INLINE_FUNCTION RealType cyl_bessel_j(const IntType& a,
+                                             const RealType& z) {
+#if defined(KOKKOS_ENABLE_CUDA)
+  KOKKOS_IF_ON_DEVICE((return jn(a, z);))
+#elif defined(__cpp_lib_math_special_functions)
+  KOKKOS_IF_ON_HOST((return std::cyl_bessel_j(a, z);))
+#endif
+  return real(cyl_bessel_j(a, Kokkos::complex<RealType>(z)));
+}
+
+template <typename RealType, typename IntType>
+KOKKOS_INLINE_FUNCTION RealType cyl_bessel_i(const IntType& a,
+                                             const RealType& z) {
+#if defined(__cpp_lib_math_special_functions)
+  KOKKOS_IF_ON_HOST((return std::cyl_bessel_i(a, z);))
+#endif
+  return real(cyl_bessel_i(a, Kokkos::complex<RealType>(z)));
+}
+
+template <typename RealType, typename IntType>
+KOKKOS_INLINE_FUNCTION RealType cyl_bessel_k(const IntType& a,
+                                             const RealType& z) {
+#if defined(__cpp_lib_math_special_functions)
+  KOKKOS_IF_ON_HOST((return std::cyl_bessel_k(a, z);))
+#endif
+  return real(cyl_bessel_k(a, Kokkos::complex<RealType>(z)));
+}
+
+template <typename RealType, typename IntType>
+KOKKOS_INLINE_FUNCTION RealType cyl_bessel_y(const IntType& a,
+                                             const RealType& z) {
+  return real(cyl_bessel_y(a, Kokkos::complex<RealType>(z)));
```

Comment:
why here we are not calling the cuda one like you did for `_j`?
I think cuda supports this too: https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__DOUBLE.html
or at least the version of the toolkit shown that

Reply:
Thanks for spotting this issue, I added it there